asset-transfer-private-data/chaincode-go/privateAssetQueries.go

```diff
@@ -0,0 +1,170 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/hyperledger/fabric-contract-api-go/contractapi"
+)
+
+
+// read asset information from collection
+func (s *SmartContract) ReadAsset(ctx contractapi.TransactionContextInterface, assetID string) (*Asset, error) {
+
+	assetJSON, err := ctx.GetStub().GetPrivateData("assetCollection", assetID) //get the asset from chaincode state
+	if err != nil {
+			return nil, fmt.Errorf("failed to read from asset %s", err.Error())
+		}
+		if assetJSON == nil {
+			return nil, fmt.Errorf("%s does not exist", assetID)
+		}
+
+		asset := new(Asset)
+	_ = json.Unmarshal(assetJSON, asset)
```

Comment:
Since this is going to be a reference sample, we should handle errors throughout.